core/src/test/java/org/jdbi/v3/core/result/TestReducing.java

```diff
@@ -64,8 +67,44 @@ public void testLeftJoinRowView() throws Exception
     }
 
     @Test
-    public void testLeftJoinResultSet() throws Exception
-    {
+    public void testCollectRows() {
+        Iterable<SomethingWithLocations> result = dbRule.getSharedHandle()
+            .createQuery("SELECT something.id, name, location FROM something NATURAL JOIN something_location")
+            .collectRows(Collector.<RowView, Map<Integer, SomethingWithLocations>, Iterable<SomethingWithLocations>>of(
+                    LinkedHashMap::new,
+                    (Map<Integer, SomethingWithLocations> map, RowView rv) ->
+                        map.computeIfAbsent(rv.getColumn("id", Integer.class),
+                                            id -> new SomethingWithLocations(rv.getRow(Something.class)))
+                           .locations
+                           .add(rv.getColumn("location", String.class)),
+                    (a, b) -> {
+                        throw new UnsupportedOperationException("shouldn't use combiner");
+                    },
+                    Map::values));
+
+        assertThat(result).containsExactly(
+            new SomethingWithLocations(new Something(1, "tree")).at("outside"),
+            new SomethingWithLocations(new Something(2, "apple")).at("tree").at("pie"));
+    }
+
+    @Test
+    public void testReduceRows() {
+        List<SomethingWithLocations> result = dbRule.getSharedHandle()
+            .createQuery("SELECT something.id, name, location FROM something NATURAL JOIN something_location")
+            .reduceRows(LinkedHashMapRowReducer.<Integer, SomethingWithLocations>of((map, rv) ->
+                map.computeIfAbsent(rv.getColumn("id", Integer.class),
+                                    id -> new SomethingWithLocations(rv.getRow(Something.class)))
+                   .locations
+                   .add(rv.getColumn("location", String.class))))
+            .collect(toList());
```

Comment:
as an example of the point above about streams, this natural invocation does a possibly unnecessary copy (especially if you use `toSet`)